OutOfSchool/OutOfSchool.WebApi/Controllers/V1/ParentController.cs

```diff
@@ -62,6 +62,31 @@ public async Task<IActionResult> Get()
             return Ok(parents);
         }
 
+        /// <summary>
+        /// Get all Parents from the database.
+        /// </summary>
+        /// <param name="filter">Filter to get a part of all parents that were found.</param>
+        /// <returns>The result is a <see cref="SearchResult{ParentDTO}"/> that contains the count of all found parents and a list of parents that were received.</returns>
+        [HasPermission(Permissions.SystemManagement)]
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(SearchResult<ParentDTO>))]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<IActionResult> GetByFilter([FromQuery] SearchStringFilter filter)
+{
+            var parents = await serviceParent.GetByFilter(filter).ConfigureAwait(false);
+
+            if (parents.TotalAmount < 1)
+            {
+                return NoContent();
```

Comment:
not exactly comment for you to fix everywhere, but a TODO/ to ask, as frontend mentioned it would be cool if the result is empty - return object with `TotalAmount` 0 and empty `Entities` list instead of `NoContent` as they need less checks for pagination.

Reply:
I did it in Parent and Provider controllers GetByFilter methods. Wrote to Sasha about it.